Recipe for swat

kesmit13 opened the pull request:
Initial recipe for SAS SWAT module

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/swat```) and found some lint.

Here's what I've got...


For **recipes/swat**:

 * The top level meta keys are in an unexpected order. Expecting ['package', 'source', 'build', 'requirements', 'test', 'about', 'extra'].
 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [21, 29]

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/swat```) and found some lint.

Here's what I've got...


For **recipes/swat**:

 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [30, 38]

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/swat```) and found it was in an excellent condition.

mariusvniekerk commented on recipes/swat/meta.yaml:
```diff
@@ -0,0 +1,63 @@
+        - pandas
+        - six
+        - requests
+        - libstdcxx-ng  # [linux64]
```
Why are these needed?  It doesn't seem as if the source on github has any c things around

mariusvniekerk replied:
I'd recommend that you add ` - compiler('cxx') ` here instead of having libgcc and libstdcxx.  
If this library ships precompiled binaries in its tarball releases this may be problematic when we switch compilers.

kesmit13 replied:
We do use pre-compiled binaries that are distributed in the tarball.  I don't know that the C dependencies are really needed since these are part of the manylinux1 spec; this package should run on most Linux distributions.  I just added them because conda build suggested it when I was writing the recipe.  I can change it to compiler('cxx') (or maybe just remove them?).

marcelotrevisani replied:
I think you can just remove that and also from run, if you really need that please add

```
  - {{ compiler('cxx') }}
  - {{ compiler('c') }}
```

into the ``build`` requirements

marcelotrevisani replied:
I think ``setuptools`` will not be necessary as well. Could you please try to remove it?

mariusvniekerk commented on recipes/swat/meta.yaml:
```diff
@@ -0,0 +1,63 @@
+    imports:
+        - swat
+
+#outputs:
```
you probably want to remove this

kesmit13 commented:
@conda-forge/staged-recipes I believe my recipe is all in order now.  I've addressed all of the suggestions and it's passing the checks.  I was just wondering if there is anything else I need to do?

marcelotrevisani commented:
LGTM

marcelotrevisani commented:
Wait.

There is some things to fix in my opinion:

```
WARNING (swat,lib/python3.6/site-packages/swat/lib/linux/t0d0en.so): Needed DSO lib/libgcc_s.so.1 found in ['libgcc-ng']
WARNING (swat,lib/python3.6/site-packages/swat/lib/linux/t0d0en.so): .. but ['libgcc-ng'] not in reqs/run, i.e. it is overlinked (likely) or a missing dependency (less likely)
WARNING (swat,lib/python3.6/site-packages/swat/lib/linux/t0d0en.so): did not find - or even know where to look for: /lib64/libcrypt.so.1
WARNING (swat,lib/python3.6/site-packages/swat/lib/linux/t0d0en.so): did not find - or even know where to look for: /lib64/libpthread.so.0
WARNING (swat,lib/python3.6/site-packages/swat/lib/linux/t0d0en.so): did not find - or even know where to look for: /lib64/libm.so.6
WARNING (swat,lib/python3.6/site-packages/swat/lib/linux/t0d0en.so): did not find - or even know where to look for: /lib64/libc.so.6
WARNING (swat,lib/python3.6/site-packages/swat/lib/linux/tk4acons.so): did not find - or even know where to look for: /usr/lib64/libstdc++.so.6
WARNING (swat,lib/python3.6/site-packages/swat/lib/linux/tk4acons.so): did not find - or even know where to look for: /lib64/libdl.so.2
```

kesmit13 commented:
I had libgcc-ng and compiler('cxx') in the recipe at different times, but the suggestion was to remove them.  As for the other libraries, those are all in the linuxmany1 list of libraries, so I would imagine most linux distributions have them.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/swat```) and found some lint.

Here's what I've got...


For **recipes/swat**:

 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [30, 38]

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/swat```) and found it was in an excellent condition.

kesmit13 commented:
I'm still not sure what needs to be done to make these other warnings go away.  When I do test builds locally conda build asks if patchelf should be run, but it was.  So I'm not sure what magic needs to be done to the recipe to make it work.

kesmit13 commented:
Here is the line from the log that runs patchelf.

```
patchelf: file: /r/ge.unx.sas.com/vol/vol510/u51/kesmit/conda-bld/swat_1529070669784/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.5/site-packages/swat/lib/linux/tkcop.so
    setting rpath to: $ORIGIN/../../../../..:$ORIGIN/../../../../../linux:/lib64/:/usr/lib64/
```

But here is the warning that I still get.

```
WARNING (swat,lib/python3.5/site-packages/swat/lib/linux/tkcop.so): /lib64/libpthread.so.0 not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/swat```) and found some lint.

Here's what I've got...


For **recipes/swat**:

 * The build section contained an unexpected subsection name. missing_dso_whitelist is not a valid subsection name.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/swat```) and found it was in an excellent condition.

kesmit13 commented:
I think since we might try to take a different direction with this package and put it on Anaconda Cloud in an organization account.  I'll close it, and re-open later if needed.
